ffi/src/schema_visitor.rs

```diff
@@ -90,43 +90,221 @@ fn visit_field_primitive_impl(
-            }
-        )*
-    };
+// TODO: turn all the primitive visitors below into a macro once cbindgen can run on macro expanded code
+///Visit a string field. Strings can hold arbitrary UTF-8 text data.
+///
+///# Safety
+///
+///Caller is responsible for providing a valid `state`, `name` slice with valid UTF-8 data,
+///and `allocate_error` function pointer.
+#[no_mangle]
+pub unsafe extern "C" fn visit_field_string(
+    state: &mut KernelSchemaVisitorState,
+    name: KernelStringSlice,
+    nullable: bool,
+    allocate_error: AllocateErrorFn,
+) -> ExternResult<usize> {
+    let name_str = unsafe { TryFromStringSlice::try_from_slice(&name) };
+    visit_field_primitive_impl(state, name_str, PrimitiveType::String, nullable)
+        .into_extern_result(&allocate_error)
+}
+
+///Visit a long field. Long fields store 64-bit signed integers.
+///
+///# Safety
+///
+///Caller is responsible for providing a valid `state`, `name` slice with valid UTF-8 data,
+///and `allocate_error` function pointer.
+#[no_mangle]
+pub unsafe extern "C" fn visit_field_long(
+    state: &mut KernelSchemaVisitorState,
+    name: KernelStringSlice,
+    nullable: bool,
+    allocate_error: AllocateErrorFn,
+) -> ExternResult<usize> {
+    let name_str = unsafe { TryFromStringSlice::try_from_slice(&name) };
+    visit_field_primitive_impl(state, name_str, PrimitiveType::Long, nullable)
+        .into_extern_result(&allocate_error)
+}
+
+///Visit an integer field. Integer fields store 32-bit signed integers.
+///
+///# Safety
+///
+///Caller is responsible for providing a valid `state`, `name` slice with valid UTF-8 data,
+///and `allocate_error` function pointer.
+#[no_mangle]
+pub unsafe extern "C" fn visit_field_integer(
+    state: &mut KernelSchemaVisitorState,
+    name: KernelStringSlice,
+    nullable: bool,
+    allocate_error: AllocateErrorFn,
+) -> ExternResult<usize> {
+    let name_str = unsafe { TryFromStringSlice::try_from_slice(&name) };
+    visit_field_primitive_impl(state, name_str, PrimitiveType::Integer, nullable)
+        .into_extern_result(&allocate_error)
+}
+
+///Visit a short field. Short fields store 16-bit signed integers.
+///
+///# Safety
+///
+///Caller is responsible for providing a valid `state`, `name` slice with valid UTF-8 data,
+///and `allocate_error` function pointer.
+#[no_mangle]
+pub unsafe extern "C" fn visit_field_short(
+    state: &mut KernelSchemaVisitorState,
+    name: KernelStringSlice,
+    nullable: bool,
+    allocate_error: AllocateErrorFn,
+) -> ExternResult<usize> {
+    let name_str = unsafe { TryFromStringSlice::try_from_slice(&name) };
+    visit_field_primitive_impl(state, name_str, PrimitiveType::Short, nullable)
+        .into_extern_result(&allocate_error)
+}
+
+///Visit a byte field. Byte fields store 8-bit signed integers.
+///
+///# Safety
+///
+///Caller is responsible for providing a valid `state`, `name` slice with valid UTF-8 data,
+///and `allocate_error` function pointer.
```

Comment:
nit: would like a space for these comments:

```suggestion
/// Visit a byte field. Byte fields store 8-bit signed integers.
///
/// # Safety
///
/// Caller is responsible for providing a valid `state`, `name` slice with valid UTF-8 data,
/// and `allocate_error` function pointer.
```